wfe/web-front-end_test.go

```diff
@@ -310,46 +419,75 @@ func TestIssueCertificate(t *testing.T) {
+				"signature":"kdu5tXk-Jz9umbi6RH-BACBj5ObJlVPA4qGsLsdbqPfn9W9CDw66Q9E1QQxt9Fxpe-fqdSDiVSfmuXhO7u068xdYptgFxWNDJXM1MH3iCs0EJz5KQ9SfGiJXrhkji_FbOdYwcxSvbThOF_qyztFmBCgZPfKKHbcGJKV3nvFDLHb6P7hIr6iqMutsFykTToYBUv3czzc87iYFpR_ukAnISLJ0hQucbMBqlinvq8TOmzi47o_uv2Fy8MF0V_C9ZYJmhZGjihhVvVlr00OaFE5bNM2uLMfr_02oG83HTjNJOGaxsqi-tuu41m3Dr5M8Ubh2oPA0OrKIuMisMMZ3aCzRtA"
+			}
+		`),
+	})
+	test.AssertEquals(t,
+		responseWriter.Body.String(),
+		"{\"type\":\"urn:acme:error:malformed\",\"detail\":\"Error creating new cert\"}")
 
-func (ra *MockRegistrationAuthority) OnValidationUpdate(authz core.Authorization) error {
-	return nil
+	responseWriter.Body.Reset()
+	wfe.NewCertificate(responseWriter, &http.Request{
+		Method: "POST",
+		Body: makeBody(`
+			{
+				"payload":"eyJhdXRob3JpemF0aW9ucyI6WyJ2YWxpZCJdLCJjc3IiOiJNSUlCR3pDQnlBSUJBREE5TVFvd0NBWURWUVFHRXdGak1Rb3dDQVlEVlFRS0V3RnZNUXN3Q1FZRFZRUUxFd0p2ZFRFS01BZ0dBMVVFQnhNQmJERUtNQWdHQTFVRUNCTUJjekJjTUEwR0NTcUdTSWIzRFFFQkFRVUFBMHNBTUVnQ1FRRFpEa3V2TllMZi1sdW1KajZyYzdoaWZiMmpxYVhxZVBYY1FhQVdGOU1YWVhDWkwwYjJRSXdRQm9ZZkp6dzA0RnNPSlFuSlpJXzBQV2MtTFMyb2FKaFRBZ01CQUFHZ0pqQWtCZ2txaGtpRzl3MEJDUTR4RnpBVk1CTUdBMVVkRVFRTU1BcUNDRzFsWlhBdVkyOXRNQXNHQ1NxR1NJYjNEUUVCQlFOQkFLVTkwcHpvMU1TbFFYZnNLWlVQTXJDcXFHWHpXWkxmY29MaHZ0Y3NfWnpDQkpscm9xYzJoUGpMQzBISXBBODNZdEFwaTlsbjIyRmlxdXdNNWVwUHZfST0ifQ",
+				"protected":"eyJhbGciOiJSUzI1NiIsImp3ayI6eyJrdHkiOiJSU0EiLCJuIjoieU5XVmh0WUVLSlIyMXk5eHNIVi1QRF9iWXdiWFNlTnVGYWw0NnhZeFZmUkw1bXFoYTd2dHR2akJfdmM3WGcyUnZnQ3hIUENxb3hnTVBUekhyWlQ3NUxqQ3dJVzJLX2tsQllOOG9ZdlR3d21lU2tBejZ1dDdaeFB2LW5aYVQ1VEpoR2swTlQya2hfelNwZHJpRUpfM3ZXLW1xeFliYkJtcHZIcXNhMV96eDlmU3VIWWN0QVpKV3p4elVaWHlrYldNV1FacEVpRTBKNGFqajUxZkluRXpWbjdWeFYtbXpmTXlib1FqdWpQaDdhTkp4QVdTcTRvUUVKSkRnV3dTaDlsZXlvSm9QcE9OSHhoNW5FRTVBakUwMUZrR0lDU3hqcFpzRi13OGhPVEkzWFhvaFVkdTI5U2UyNmsyQjBQb2xEU3VqMEdJUVU2LVc5VGRMWFNqQmIyU3BRIiwiZSI6IkFBRUFBUSJ9fQ",
+				"signature":"n7JIc5QeR0PK1JyIOIdJ_KWwNwtz2Ke89FGbnAb3LRbpSbkMnZNx1TNeg5TNxSLGv9FSoCcmNvMlmK5DR1sRJddsnryI1F36qZl6lvv-ZrVjKlJ6__HiQCXdVsXGo5hMNLc834cyp1RwadmjQr46cv5vv0cOSuPfSsjdNC9Aqw7AqVtqOT6ONmZQdwQFKVxK8I34wx67-pc4Rwia5Bbo6yZFebgKQkDc3EzJsqeUHY5aTbCK6aD9wKx4PA5M3bwVy5xTPNsUAsSYHdxvnTQ9WpzY20f_7ha8Zdw8Xf2y6pZ1F8mjGHAifqTzEMZpHTjW8-6syiq9nSWUM4KiIODkdA"
+			}
+		`),
+	})
+	randomCertDer, _ := hex.DecodeString(RandomCert)
+	test.AssertEquals(t,
+		responseWriter.Body.String(),
+		string(randomCertDer))
+	test.AssertEquals(
+		t, responseWriter.Header().Get("Location"),
+		"/acme/cert/0000000000000000")
+	test.AssertEquals(
+		t, responseWriter.Header().Get("Link"),
+		"</acme/issuer-cert>;rel=\"up\"")
+	test.AssertEquals(
+		t, responseWriter.Header().Get("Content-Type"),
+		"application/pkix-cert")
 }
 
 func TestChallenge(t *testing.T) {
-	wfe := NewWebFrontEndImpl()
+	// wfe := NewWebFrontEndImpl()
```

Comment:
Don't just comment stuff out.  Fix or delete.  See above comment about re-instantiating the WFE for each test.